### PR TITLE
Add a data_format flag to the models + other fixes

### PR DESCRIPTION
**Pull request by nealwu**

### CODE REVIEW

File: official/mnist/mnist.py [hunk 1]
```diff
@@ -35,8 +35,19 @@
 parser.add_argument('--model_dir', type=str, default='/tmp/mnist_model',
                     help='The directory where the model will be stored.')
 
-parser.add_argument('--steps', type=int, default=20000,
-                    help='Number of steps to train.')
+parser.add_argument('--train_epochs', type=int, default=40,
+                    help='Number of epochs to train.')
+
+parser.add_argument(
+    '--data_format', type=str, default=None,
```
Comment (k-w-w) [inline official/mnist/mnist.py]: Should we set a default here? If not, then the README needs to be updated

File: official/resnet/imagenet_main.py [hunk 2]
```diff
@@ -53,6 +53,12 @@
     '--batch_size', type=int, default=32,
     help='Batch size for training and evaluation.')
 
+parser.add_argument(
+    '--data_format', type=str, default=None,
```
Comment (k-w-w) [inline official/resnet/imagenet_main.py]: Same here

File: official/mnist/mnist.py (hunk 1 above)
Comment (nealwu) [reply official/mnist/mnist.py]: The default is `None`, in which case it sets `data_format` based on a check to `tf.test.is_built_with_cuda()`. This is just an option to override that (e.g., run with NHWC even though you have a GPU).

Comment (nealwu) [reply official/mnist/mnist.py]: I can add to the help message to explain the behavior when the flag is unspecified.

Comment (nealwu) [reply official/mnist/mnist.py]: Added.

Comment (k-w-w) [reply official/mnist/mnist.py]: Ohhh I see! Thanks for the changes

**Pull request merged**